modules/integration/tests-integration/src/test/java/org/wso2/jaggery/integration/tests/hostObjects/core/FileHostObjectTestCase.java

```diff
@@ -35,139 +36,144 @@
  */
 public class FileHostObjectTestCase {
 
-    @Test(groups = {"jaggery"},
-          description = "Test for file host object")
-    public void testFileExist() {
+    private static final Log log = LogFactory.getLog(FileHostObjectTestCase.class);
+
+    @Test(groups = { "jaggery" }, description = "Test for file host object")
+    public void testFileExist() throws IOException {
+        ClientConnectionUtil.waitForPort(9763);
+        String finalOutput = null;
+        BufferedReader in = null;
+        try {
+            URL jaggeryURL = new URL("http://localhost:9763/testapp/file.jag");
+            URLConnection jaggeryServerConnection = jaggeryURL.openConnection();
+            in = new BufferedReader(new InputStreamReader(jaggeryServerConnection.getInputStream()));
+            String inputLine;
+            if(in != null) {
+                while ((inputLine = in.readLine()) != null) {
+                    finalOutput = inputLine;
+                }
+            }
+        } catch (IOException e) {
+            log.error("Reading InputStream fails in fileExist test method", e);
+        } finally {
+            in.close();
```

Comment:
If there is a error while opening the file, then there will be an exception. In that case inputstream - 'in' will be null, and hence this statement will cause NPE. Please do a null check before you close the stream.